metrics tracer implementation

ravjotbrar opened the pull request:
> Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

## Description

> Please provide a detailed description for the change.
> As much as possible, please try to keep changes separate by purpose. For example, try not to make a one-line bug fix in a feature request, or add an irrelevant README change to a bug fix.

## Impact

> What's the impact of this change?

## Testing

> Have you added unit and integration tests if necessary?
> Were any tests changed? Are any breaking changes necessary?

## Additional Information

> Any additional details that we should be aware of?

## Checklist

- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-spanner/issues/new/choose) before writing your code! That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease
- [ ] Appropriate docs were updated
- [ ] Appropriate comments were added, particularly in complex areas or places that require background
- [ ] No new warnings or issues will be generated from this change

Fixes #issue_number_goes_here 🦕

ravjotbrar committed:
chore: Add Custom OpenTelemetry Exporter in for Service Metrics

ravjotbrar committed:
Add readme

lszinv committed:
removed unused new packages

lszinv committed:
Added headers

ravjotbrar committed:
Address review comments

ravjotbrar committed:
Address review comments

ravjotbrar committed:
Remove env var from .mocharc.js

odeke-em committed:
chore: add x_goog_spanner_request_id as attribute to spans (#2275)

This change adds the x-goog-spanner-request-id value that is
sent as a gRPC header, but as a span attribute with the
key `x_goog_spanner_request_id` to aid in better debugging
and correlation.

Updates #2200

surbhigarg92 committed:
Merge branch 'main' into custom_exporter

release-please[bot] committed:
chore(main): release 7.20.0 (#2253)

Co-authored-by: release-please[bot] <55107282+release-please[bot]@users.noreply.github.com>

surbhigarg92 committed:
Merge branch 'main' into custom_exporter

ravjotbrar committed:
Add grpc-js package

ravjotbrar committed:
metrics tracer implementation

lszinv commented on package.json:
```diff
@@ -84,7 +85,8 @@
     "teeny-request": "^9.0.0",
-    "through2": "^4.0.0"
+    "through2": "^4.0.0",
+    "ts-node": "^10.9.2"
```
This is weird to have as a dependency. At best, it should be a dev dependency

lszinv committed:
Fix tests

gcf-owl-bot[bot] committed:
chore: do not run windows tests in the test-script environment (#2273)

Source-Link: https://togithub.com/googleapis/synthtool/commit/490078a0bf0777f7f4400b67b6ae77d3682aae42
Post-Processor: gcr.io/cloud-devrel-public-resources/owlbot-nodejs:latest@sha256:7c7ecb30ffac77ebdacd15f4b1c6c888f01c212832b9efd73fbf4bfc1284b7d4

gcf-owl-bot[bot] committed:
chore: update owlbot-nodejs dependencies (#2276)

* chore: update owlbot-nodejs dependencies

* Update container_test.yaml

Source-Link: https://togithub.com/googleapis/synthtool/commit/1e798e6de27c63a88a1768c2a5f73b85e1523a21
Post-Processor: gcr.io/cloud-devrel-public-resources/owlbot-nodejs:latest@sha256:c7e4968cfc97a204a4b2381f3ecb55cabc40c4cccf88b1ef8bef0d976be87fee

aksharauke committed:
feat: Adding sample for pre-split feature (#2274)

* feat: Adding sample for pre-split feature

* build fixes

* lint fixes

* more lint fixes

* more lint fixes

* more lint fixes

* more lint fixes

* fixed lint

* review comments addressed

* linter and positional args changes

* linter fixes

* corrected the positional params

* build error fix

* lint fixes

* lint fixes

---------

Co-authored-by: surbhigarg92 <surbhigarg.92@gmail.com>

surbhigarg92 committed:
fix: adding span attributes for request tag and transaction tag (#2236)

surbhigarg92 committed:
Merge branch 'main' into custom_exporter

release-please[bot] committed:
chore(main): release 7.21.0 (#2281)

* chore(main): release 7.21.0

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

---------

Co-authored-by: release-please[bot] <55107282+release-please[bot]@users.noreply.github.com>
Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>

surbhigarg92 committed:
Merge branch 'main' into custom_exporter

alkatrivedi committed:
chore: remove test(14) and test(16) from the required checks (#2283)

* chore: remove test(14) and test(16) from the required checks

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

* exclude sync-repo-settings.yaml from owlbot

---------

Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>

ravjotbrar committed:
Merge branch 'main' into custom_exporter

ravjotbrar committed:
merge conflicts with exporter changes
